app/assets/javascripts/lib/views/d3_chart_view.coffee

```diff
@@ -26,8 +26,15 @@ class @D3ChartView extends BaseChartView
       @check_merit_enabled()
       @drawn = App.settings.merit_order_enabled()
 
+    @display_empty_message()
     @refresh()
 
+  is_empty: =>
+    false
+
+  display_empty_message: =>
+    EmptyChartMessage.display.call(this)
```

Comment:
Same with `template`. I think having to `call` and `apply` is really confusing (and easy-to-forget), when simply passing in a parameter will do just as well...